### PR TITLE
PQSDKd plugin implementation

### PR DESCRIPTION
**Pull request by henrydcase**

### CODE REVIEW

File: src/libstrongswan/plugins/pqsdk/pqsdkd/Makefile.am [hunk 1]
```diff
@@ -0,0 +1,23 @@
+AM_CPPFLAGS = \
+	-I$(top_srcdir)/src/libstrongswan
+
+AM_CFLAGS = \
+	$(PLUGIN_CFLAGS) -I/opt/pqsdk/includes
```
Comment (henrydcase) [inline src/libstrongswan/plugins/pqsdk/pqsdkd/Makefile.am:5]: ``./configure`` should check if pqsdkd is installed

File: src/libstrongswan/plugins/pqsdk/pqsdkd/pqsdkd_plugin.c [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+
+    if (plugin->is_pqsdk_on) {
+        // already initialized
+        goto end;
+    }
+
+    plugin->comm.socket_path = lib->settings->get_str(lib->settings,
+        "%s.plugins.pqsdk-pqsdkd.socket_path", NULL, lib->ns);
+
+	if (!plugin->comm.socket_path) {
```
Comment (OussamaDanba) [inline src/libstrongswan/plugins/pqsdk/pqsdkd/pqsdkd_plugin.c:68]: These lines are indented using tabs while the rest isn't. A little later this happens too.

File: src/libstrongswan/plugins/pqsdk/pqsdkd/kem.c [hunk 3]
```diff
@@ -0,0 +1,345 @@
...
+}
+
+/**
+ * Send req.len bytes to the c.req and read rsp.len just after.
+ * This function will block.
+ */
+static void exchng(chunk_t rsp, const struct comm_t *c, const chunk_t req) {
+    size_t t = 0;
+    while (t < req.len) {
+        t += write(c->req, &req.ptr[t], req.len - t);
```
[the remark this answers is not in the document]
Comment (henrydcase) [reply src/libstrongswan/plugins/pqsdk/pqsdkd/kem.c]: this part will be rewritten after we switch to daemon as a server.
but anyway that's fixed now. Thanks!

**Comment by henrydcase** [comment]
thanks for checking!

**Pull request merged**